File: src/routes/status.js
/**
 * @swagger
 * tags:
 *   name: Status
 *   description: API health and system status
 */

const express = require('express');
const eslService = require('../services/eslService');
const dbService = require('../services/dbService');
const wsService = require('../services/wsService');
const { authenticate } = require('../middleware/auth');

const router = express.Router();

/**
 * @swagger
 * /api/status:
 *   get:
 *     summary: API health check (no auth required)
 *     tags: [Status]
 *     responses:
 *       200:
 *         description: API is running
 */
router.get('/', (req, res) => {
  res.json({
    status: 'ok',
    service: 'FusionPBX API Bridge',
    version: '1.0.0',
    timestamp: new Date().toISOString(),
  });
});

/**
 * @swagger
 * /api/status/detailed:
 *   get:
 *     summary: Detailed system status (requires auth)
 *     tags: [Status]
 *     security:
 *       - ApiKeyAuth: []
 *       - BearerAuth: []
 *     responses:
 *       200:
 *         description: System status details
 */
router.get('/detailed', authenticate, async (req, res, next) => {
  try {
    const [dbOk] = await Promise.all([dbService.testConnection()]);
    const eslStatus = eslService.getStatus();
    const wsClients = wsService.getConnectedClients();

    res.json({
      status: 'ok',
      timestamp: new Date().toISOString(),
      services: {
        esl: { ...eslStatus },
        database: { connected: dbOk },
        websocket: { connectedClients: wsClients.length, clients: wsClients },
      },
    });
  } catch (err) {
    next(err);
  }
});

module.exports = router;

File: python/routers/status.py
from datetime import datetime
from fastapi import APIRouter, Depends
from deps.auth import require_auth
from services.esl_service import esl_service
from services.db_service import db_service
from services.ws_service import ws_service

router = APIRouter(prefix='/api/status', tags=['Status'])


@router.get('/')
async def status():
    return {
        'status': 'ok',
        'service': 'FusionPBX API Bridge',
        'version': '1.0.0',
        'timestamp': datetime.utcnow().isoformat() + 'Z',
    }


@router.get('/detailed')
async def status_detailed(_user=Depends(require_auth)):
    db_ok = await db_service.test_connection()
    esl_status = esl_service.get_status()
    ws_clients = ws_service.get_connected_clients()
    return {
        'status': 'ok',
        'timestamp': datetime.utcnow().isoformat() + 'Z',
        'services': {
            'esl': esl_status,
            'database': {'connected': db_ok},
            'websocket': {'connectedClients': len(ws_clients), 'clients': ws_clients},
        },
    }
